### PR TITLE
Update metric values in backup manager

### PR DESCRIPTION
**Pull request by mgao0**
This PR is second commit of zk backup monitoring implementation. This commit integrates BackupStats with BackupManager.

Test: BackupBeanTest

### CODE REVIEW

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java [hunk 1]
```diff
@@ -63,9 +64,12 @@
   private final BackupStorageProvider backupStorage;
   private final long serverId;
   private final String namespace;
-  private BackupBean backupBean = null;
+  @VisibleForTesting
+  protected BackupBean backupBean = null;
   private BackupStats backupStats = null;
 
+  private int snapshotFilesBackedUp;
```
Comment (narendly) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java]: Nit: `numSnapshotFilesBackedUp`?

Also, should this be inside `backupStats`? Is there a reason we're giving this the scope inside `BackupManager` and outside `BackupStats`?

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/monitoring/BackupStats.java [hunk 2]
```diff
@@ -82,6 +85,32 @@ public long getNumberOfSnapshotFilesBackedUpLastIteration() {
...
+    lastSnapshotBackupIterationStartTime = System.currentTimeMillis();
+    snapshotBackupActive = true;
+  }
+
+  /**
+   *  Record the status and timestamp when a snapshot backup iteration finishes
+   * @param errorFree If this iteration finishes without error
+   * @param snapshotBackedUp How many snapshot files are backed up within the iteration
+   */
+  public void setSnapshotBackupIterationDone(boolean errorFree, int snapshotBackedUp) {
```
Comment (narendly) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/monitoring/BackupStats.java]: `snapshotBackedUp` -> `numSnapshotFilesBackedUp`

File: zookeeper-server/src/test/java/org/apache/zookeeper/server/backup/BackupBeanTest.java [hunk 3]
```diff
@@ -157,4 +158,117 @@ private boolean containsMBean(Set<ZKMBeanInfo> mbeanSet, String mbeanName, boole
...
+    Assert.assertEquals(1, backupBean.getNumConsecutiveFailedSnapshotIterations());
+    Assert.assertEquals(2, backupBean.getNumConsecutiveFailedTxnLogIterations());
+
+    createNode(connection, nodeNames[4]);
+    backupManager.getSnapBackup(snapLog, true).run(1);
+    backupManager.getLogBackup(snapLog, true).run(1);
+    Assert.assertEquals(0, backupBean.getNumConsecutiveFailedSnapshotIterations());
+    Assert.assertEquals(0, backupBean.getNumConsecutiveFailedTxnLogIterations());
+
+    Thread.sleep(60 * 1000);
```
Comment (narendly) [inline zookeeper-server/src/test/java/org/apache/zookeeper/server/backup/BackupBeanTest.java:208]: Nit: `L` for longs

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java (hunk 1 above)
Comment (mgao0) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java]: This is inside`backupStats`, it's the `getNumberOfSnapshotFilesBackedUpLastIteration` metric. The field here is for calculation purpose, so within one iteration we can increment this number every time one more snapshot file is backed up and we can report this number to the `getNumberOfSnapshotFilesBackedUpLastIteration` metric and reset this number when an iteration ends.

File: zookeeper-server/src/test/java/org/apache/zookeeper/server/backup/BackupBeanTest.java (hunk 3 above)
Comment (mgao0) [reply zookeeper-server/src/test/java/org/apache/zookeeper/server/backup/BackupBeanTest.java:208]: `backupBean.getNumConsecutiveFailedTxnLogIterations()` in an int typed variable.

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java (hunk 1 above)
Comment (narendly) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java]: What I meant is that the scope of this internal variable lies within backupStats (conceptually) - BackupManager does not need to know this so it doesn't need access to it. You could add helper functions like `incrementNumSnapshotFilesBackedUp()` to update `BackupStats`'s internal data.

Comment (mgao0) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java]: Good point! Updated the code.

**Review by narendly — APPROVED** [review]

**Pull request merged**